packages/syft/src/syft/core/tensor/smpc/mpc_tensor.py

```diff
@@ -352,6 +353,78 @@ def __getattribute__(self, attr_name: str) -> Any:
 
         return object.__getattribute__(self, attr_name)
 
+    @staticmethod
+    def reshare(self: MPCTensor, parties: Iterable[Any]) -> MPCTensor:
+        """Reshare a given secret to a superset of parties.
+
+        Args:
+            parties(List[Any]): Input parites List.
+
+        Returns:
+            res_mpc(MPCTensor): Reshared MPCTensor.
+
+        Raises:
+            ValueError: If the input MPCTensor and input parties are same.
+        """
+        mpc_parties = set(self.parties)
+        parties = set(parties)
+        shape = self.shape
+        seed_shares = self.seed_shares
+        client_map = {share.client: share for share in self.child}
+        if mpc_parties == parties:
+            raise ValueError(
+                "Input parties for resharing are same as the input parties."
+            )
+
+        shares = [client_map.get(party) for party in parties]
+        for i, party in enumerate(parties):
+            shares[
+                i
+            ] = party.syft.core.tensor.smpc.share_tensor.ShareTensor.generate_przs(
+                rank=i,
+                nr_parties=len(parties),
+                value=shares[i],
+                shape=shape,
+                seed_shares=seed_shares,
+            )
+
+        res_mpc = MPCTensor(shares=shares, shape=shape, parties=parties)  # type: ignore
+
+        return res_mpc
+
+    @staticmethod
+    def sanity_checks(self: MPCTensor, other: Any) -> Tuple[MPCTensor, Any]:
+        """Performs sanity checks to share data to whole superset of parites involved.
+
+        Args:
+            other (Any): input operand.
+
+        Returns:
+            Tuple[MPCTensor,Any]: Rehared Tensor values.
+        """
+        if ispointer(other):
+            parties = self.parties
+            client = other.client
+            public_shape = other.public_shape
+            if public_shape is None:
+                # TODO: Should be modified after Trask's Synthetic data PR.
+                raise ValueError("The input tensor pointer should have public shape.")
+            if client not in parties:
+                parties.append(client)
+                self = MPCTensor.reshare(self, parties)
+
+            other = MPCTensor(secret=other, parties=parties, shape=public_shape)
+
+        elif isinstance(other, MPCTensor):
+            p1 = set(self.parties)  # parties in first MPCTensor
```

Comment:
BUG?: What happens if we have:
```
self.parties = [p1, p2]
other.parties = [p2, p1]
```
The sets would be equal, right?

But in that case - if we were to do a `private_add` we would have: share_p1 + share_p2 <-- will throw an exception because they are not on the same machine?

Reply:
We sort the MPCTensor child nodes, every time we instantiate it george,
https://github.com/OpenMined/PySyft/blob/a7d5ac43534f96b9586fadf7302ddca5b0554d18/packages/syft/src/syft/core/tensor/smpc/mpc_tensor.py#L107
by which we call on the correct mpc tensors,

Reply:
maybe I will also include a sanity check in ops level for robust code.